src/App.jsx

```diff
@@ -1,248 +1,303 @@
-                    </span>
-                  </a>
-                </span>
-              </th>
-
-              <th>
-                <span className="is-flex is-flex-wrap-nowrap">
-                  User
-
-                  <a href="#/">
-                    <span className="icon">
-                      <i data-cy="SortIcon" className="fas fa-sort" />
-                    </span>
-                  </a>
-                </span>
-              </th>
-            </tr>
-          </thead>
-
-          <tbody>
-            <tr data-cy="Product">
-              <td className="has-text-weight-bold" data-cy="ProductId">
-                1
-              </td>
-
-              <td data-cy="ProductName">Milk</td>
-              <td data-cy="ProductCategory">🍺 - Drinks</td>
-
-              <td
-                data-cy="ProductUser"
-                className="has-text-link"
-              >
-                Max
-              </td>
-            </tr>
+function getCategoryById(categoryId) {
+  return categoriesFromServer.find(category => category.id === categoryId);
+}
+
+function getUserById(userId) {
+  return usersFromServer.find(user => user.id === userId)
+}
+
+const owners = categoriesFromServer.map(own => ({
+  user: getUserById(own.ownerId)
+}));
+
+function getOwnUsersById(userId) {
+  return owners.filter(ownUser => ownUser.user.id === userId)
+    || null;
+}
+
+const products = productsFromServer.map(product => ({
+  ...product,
+  category: getCategoryById(product.categoryId),
+  user: getOwnUsersById(product.userId)
+}));
+
+
+function getFilterProducts(products, query) {
+  let productFilter = [...products];
+
+  if (query) {
+    const normalizeQuery = query.toLowerCase().trim();
+
+    productFilter = productFilter.filter(product => (
+      product.name.toLowerCase().startsWith(normalizeQuery)
+    ));
+  }
+
+  return productFilter;
+}
+
+console.log(products)
+
+export const App = () => {
+  // const category = products.map(category => category.name)
+  // console.log(category)
+  // const [sorted, setSorted] = useState(category)
+  const [select, setSelected] = useState('')
+  const [query, setQuery] = useState('');
+
+  const productForRendering = getFilterProducts(products, query);
+
+  const sortByGrocery = () => {
+    setSelected('Grocery')
+  }
+
+  const sortByDrinks = () => {
+    setSelected('Drinks')
+  }
+
+  const sortByFruits = () => {
+    setSelected('Fruits')
+  }
+
+  const sortByElectronics = () => {
+    setSelected('Electronics')
+  }
+
+  const sortByClothes = () => {
+    setSelected('Clothes')
+  }
+
+
+  const deleteBtn = () => {
+    setQuery('')
+  }
 
-            <tr data-cy="Product">
-              <td className="has-text-weight-bold" data-cy="ProductId">
-                2
-              </td>
+  const sortByAll = () => {
+    setSelected('All')
+  }
 
-              <td data-cy="ProductName">Bread</td>
-              <td data-cy="ProductCategory">🍞 - Grocery</td>
+  const sortByRoma = () => {
```

Comment:
Also you need to add a state with user id and then filter products by user id instead of writing every function for each user because there can be a lot of users